src/components/ui/modal/Modal.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect } from 'react';
+import { useModal } from 'hooks/useModal';
+import { useNavigate } from 'react-router-dom';
+
+export const Modal = ({
+   children,
+   isShowing,
+   target,
+   disableCancle,
+}: {
+   children: React.ReactNode;
+   isShowing: boolean;
+   target?: string;
+   disableCancle?: boolean;
+}) => {
+   const { open } = useModal();
+   const navigate = useNavigate();
+
+   const alertOption = ({ target, disableCancle }: { target?: string; disableCancle?: boolean }) => {
+      return {
+         title: '알림',
+         accept: {
+            text: '확인',
+            onClick: () => {
+               target && navigate(target);
+            },
+         },
+         disableCancle,
+      };
+   };
+
+   useEffect(() => {
+      isShowing && open(children, alertOption({ target, disableCancle }));
+   }, [isShowing]);
+
+   return <></>;
```

Comment:
컴포넌트를 분리하라고 말씀 드렸던 게 <Modal> 자체를 렌더링하는 컴포넌트가 있었으면 좋겠다는 의미였는데 해당 컴포넌트는 provider와 모달 사용부를 연결해주는 중간다리 역할을 하고 있군요! 이 경우는 jsx를 렌더링하고 있지 않기 때문에 hook 형태로 만드는 게 조금 더 적합할 것 같습니다. provider 내부에 해당 코드들을 관리하는 것도 괜찮을 것 같구요!!
```suggestion
   return <></>;
```